src/funding/funding.ts

```diff
@@ -79,27 +78,29 @@ export class Funding {
     this.dealId = navigationInstruction.params.address;
     await this.dealService.ensureInitialized();
     this.deal = this.dealService.deals.get(this.dealId);
+    //wait until the deal data is there
     await this.deal.ensureInitialized();
     //Make sure the connected account is part of this deal. Otherwise redirect to home page.
     this.verifySecurity();
+    //wait until the dao transactions from the contract are there
+    await Utils.waitUntilTrue(() => this.deal.daoTokenTransactions !== undefined);
   }
 
   public async bind(): Promise<void> {
-    if (!this.deal.daoRepresentedByCurrentAccount) this.verifySecurity();
     //get contract token information from the other DAO
     //Clone the tokens from registration data and add props from ITokenFunding
-    this.daoOtherThanRepresentedByCurrentAccountTokens = JSON.parse(JSON.stringify(this.deal.daoOtherThanRepresentedByCurrentAccount.tokens));
-    this.daoOtherThanRepresentedByCurrentAccountTokens.forEach(x => {this.setTokenContractInfo(x, this.deal.daoOtherThanRepresentedByCurrentAccount);});
+    this.secondDaoTokens = JSON.parse(JSON.stringify(this.secondDao.tokens));
+    this.secondDaoTokens.forEach(x => {this.setTokenContractInfo(x, this.secondDao);});
 
     //TODO figure out what the vested amount is from the deal
     //this.vestedAmount = this.deal.vestedAmount;
     this.vestedAmount = 0;
 
     //get contract token information from the DAO related to the account
-    this.daoRepresentedByCurrentAccountTokens = JSON.parse(JSON.stringify(this.deal.daoRepresentedByCurrentAccount.tokens));
-    this.daoRepresentedByCurrentAccountTokens.forEach(x => {this.setTokenContractInfo(x, this.deal.daoRepresentedByCurrentAccount);});
+    this.firstDaoTokens = JSON.parse(JSON.stringify(this.firstDao.tokens));
```

Comment:
optional: this cloneDeep seems useful in general
--> Utils.cloneDeep

Reply:
@vipergtsrz - Check if the native JS function exists and use that else the above.

Reply:
I added a new cloneDeep method and changed my code to call that. It uses the new built in structuredClone method if it's supported and if it's not it falls back to JSON.parse(JSON.stringify())